app/assets/javascripts/discourse/app/components/site-header.js

```diff
@@ -276,6 +253,27 @@ const SiteHeaderComponent = MountWidget.extend(
       });
     },
 
+    _stickyHeaderCheck() {
+      const dockAnchor = document.querySelector(".header-dock-anchor");
+      if (dockAnchor) {
+        this._dockAnchor = dockAnchor;
```

Comment:
If we only use this _dockAnchor for cleanDom, I'm not sure there's big value in keeping it around, maybe we just requery dom when cleaning it